source/auth.js

```diff
@@ -9,7 +10,67 @@ function generateTokenAuthHeader(tokenInfo) {
     return `${tokenInfo.token_type} ${tokenInfo.access_token}`;
 }
 
+function generateDigestAuthHeader(options, digest) {
+    function md5(data) {
+        return crypto
+            .createHash("md5")
+            .update(data)
+            .digest("hex");
+    }
+
+    function ha1Compute(algorithm, user, realm, pass, nonce, cnonce) {
```

Comment:
Could we take these functions out of `generateDigestAuthHeader` and put them on the top level? It'd be nice to add tests for them at some stage :)

Reply:
`md5` could be taken out too, even though it's quite basic.

Reply:
I have moved them to a separate `crypto.js` file, in the same fashion you have a separate `encode.js` file.

(FYI, I amended previous commit only to fix its author/committer.)